libraries/networking/src/ExternalResource.h

```diff
@@ -0,0 +1,139 @@
+     * Returns the location of the resource \p relative_path in bucket \p bucket
+     *
+     * @note The resulting path will be sanitized by condensing multiple instances of '/' to one.
+     * This is done for easier usage with Amazon S3 and compatible systems.
+     *
+     * @param bucket The bucket in which the resource is found
+     * @param relative_path The path of the resource within the bucket
+     * @returns The resulting URL as a QUrl
+     */
+    Q_INVOKABLE QUrl getQUrl(Bucket bucket, const QUrl& relative_path);
+
+
+    /**
+     * Returns the location of a resource as a QUrl
+     *
+     * Returns the location of the resource \p relative_path in bucket \p bucket
+     *
+     * @note The resulting path will be sanitized by condensing multiple instances of '/' to one.
+     * This is done for easier usage with Amazon S3 and compatible systems.
+     *
+     * @param bucket The bucket in which the resource is found
+     * @param relative_path The path of the resource within the bucket
+     * @returns The resulting URL as a QUrl
+     */
+    Q_INVOKABLE QUrl getQUrl(Bucket bucket, QString path) {
+        return getQUrl(bucket, QUrl(path));
+    }
+
+    /**
+     * Returns the location of a resource as a QString
+     *
+     * Returns the location of the resource \p relative_path in bucket \p bucket
+     *
+     * @note The resulting path will be sanitized by condensing multiple instances of '/' to one.
+     * This is done for easier usage with Amazon S3 and compatible systems.
+     *
+     * @param bucket The bucket in which the resource is found
+     * @param relative_path The path of the resource within the bucket
+     * @returns The resulting URL as a QString
+     */
+    Q_INVOKABLE QString getUrl(Bucket bucket, const QUrl &relative_path) {
+        return ExternalResource::getQUrl(bucket, relative_path).toString();
+    };
+
+    /**
+     * Returns the location of a resource as a QString
+     *
+     * Returns the location of the resource \p relative_path in bucket \p bucket
+     *
+     * @note The resulting path will be sanitized by condensing multiple instances of '/' to one.
+     * This is done for easier usage with Amazon S3 and compatible systems.
+     *
+     * @param bucket The bucket in which the resource is found
+     * @param relative_path The path of the resource within the bucket
+     * @returns The resulting URL as a QString
+     */
+    Q_INVOKABLE QString getUrl(Bucket bucket, const QString &relative_path) {
+        return ExternalResource::getQUrl(bucket, QUrl(relative_path)).toString();
+    };
+
+    private:
+    ExternalResource(QObject* parent = nullptr);
+
+    QMap<Bucket, QUrl> _bucketBases {
+        { Bucket::Public, QUrl("https://public.vircadia.com")},
+        { Bucket::Content, QUrl("https://content.vircadia.com")},
+        { Bucket::Assets, QUrl("https://assets.vircadia.com")},
+        { Bucket::MPAssets, QUrl("https://mpassets.vircadia.com")},
+    };
+};
+
+#endif
```

Comment:
some general code style things (beating ctrlaltdavid to the punch):

- extra blank lines in the file above
- remove space before/after parens in if statements
- unindent public/private
- newline at the end of this file
- spaces after `;` below

Reply:
Ran the class through the formatter